packages/player/src/components/video.tsx

```diff
@@ -51,33 +53,38 @@ export const Video: React.FC<VideoProps> = ({
   //to the videoService and update our state machine context
   const videoElemRef = React.useRef<any>(null)
   const videoService = useVideo()
+  const paused = useSelector(videoService, selectIsPaused)
 
   useMetadataTrackList()
 
+  React.useEffect(() => {
+    videoService.send({type: 'REGISTER', videoRef: videoElemRef})
+
+    // sometimes the event handlers aren't registered before the
+    // `canPlay` event is fired (caching, for instance) so we want
+    // to check and see if it's ready as soon as we get a ref to
+    // the HTMLVideoElement and "manually" fire an event
+    if (videoElemRef.current && videoElemRef.current.readyState > 3) {
+      videoService.send('LOADED')
+    }
+  }, [videoElemRef.current])
+
   return (
     <video
       className={cx([`cueplayer-react-video`, className])}
       id={id}
       crossOrigin={crossOrigin}
-      ref={(c: HTMLVideoElement) => {
-        videoElemRef.current = c
-        videoService.send({type: 'REGISTER', videoRef: videoElemRef})
-
-        // sometimes the event handlers aren't registered before the
-        // `canPlay` event is fired (caching, for instance) so we want
-        // to check and see if it's ready as soon as we get a ref to
-        // the HTMLVideoElement and "manually" fire an event
-        if (c && c.readyState > 3) {
-          videoService.send('LOADED')
-        }
-      }}
+      ref={videoElemRef}
```

Comment:
this fixes following error I was getting when using `videoElemRef`
![image](https://user-images.githubusercontent.com/25487857/151168932-6b861a26-2558-499f-895c-c467eb379a81.png)

I reckon It was caused by passing an inline arrow as a ref, which would go through unset / set cycle on every render ([read more](https://thoughtspile.github.io/2021/05/17/everything-about-react-refs/)).